### PR TITLE
enhance(chips): Contact Chips

### PR DESCRIPTION
**Pull request by typotter**
Many style updates and changes
Contact Chips component (comprising chips and autocomplete)
refactoring of ChipsDirective and user templates using Rob's Awesome Technique (patent pending)

### CODE REVIEW

File: src/components/chips/js/contactChipsDirective.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   *
+   * @usage
+   * <hljs lang="html">
+   *   <md-contact-chips
+   *       ng-model="ctrl.contacts"
+   *       md-contacts="ctrl.querySearch($query)"
+   *       md-contact-name="name"
+   *       md-contact-image="image"
+   *       md-contact-email="email"
+   *       filter-selected="ctrl.filterSelected"
```
Comment (ThomasBurleson) [inline src/components/chips/js/contactChipsDirective.js]: This should be `md-filter-selected` right ?

File: src/components/chips/contact-chips.spec.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
+describe('<md-contact-chips>', function() {
+  var scope;
+  var CONTACT_CHIPS_TEMPLATE = '\
+      <md-contact-chips\
+          ng-model="contacts"\
+          md-contacts="querySearch($query)"\
+          md-contact-name="name"\
+          md-contact-image="image"\
```
Comment (ThomasBurleson) [inline src/components/chips/contact-chips.spec.js]: This is generating a 404 in the Karma Webserver.
Can you please do hardcoded data image in the ng-src; @see https://css-tricks.com/data-uris/

e.g
```
data:image/gif;base64,R0lGODlhEAAQAMQAAORHHOVSKudfOulrSOp3WOyDZu6QdvCchPGolfO0o/XBs/fNwfjZ0frl3/zy7////wAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACH5BAkAABAALAAAAAAQABAAAAVVICSOZGlCQAosJ6mu7fiyZeKqNKToQGDsM8hBADgUXoGAiqhSvp5QAnQKGIgUhwFUYLCVDFCrKUE1lBavAViFIDlTImbKC5Gm2hB0SlBCBMQiB0UjIQA7
```

File: src/components/chips/js/contactChipsDirective.js (hunk 1 above)
Comment (typotter) [inline src/components/chips/js/contactChipsDirective.js]: yup. good catch

File: src/components/chips/contact-chips.spec.js (hunk 2 above)
Comment (typotter) [inline src/components/chips/contact-chips.spec.js]: done

File: src/components/chips/js/contactChipsDirective.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+   *    displayed when there is at least on item in the list
+   * @param {expression} md-contacts An expression expected to return contacts matching the search
+   *    test, `$query`.
+   * @param {string} md-contact-name The field name of the contact object representing the
+   *    contact's name.
+   * @param {string} md-contact-email The field name of the contact object representing the
+   *    contact's email address.
+   * @param {string} md-contact-image The field name of the contact object representing the
+   *    contact's image.
+   * @param {expression=} filter-selected Whether to filter selected contacts from the list of
```
Comment (ThomasBurleson) [inline src/components/chips/js/contactChipsDirective.js]: should this be `md-filter-selected`

File: src/components/chips/js/contactChipsDirective.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      bindToController: true,
+      compile: compile,
+      scope: {
+        contactQuery: '&mdContacts',
+        placeholder: '@',
+        secondaryPlaceholder: '@',
+        contactName: '@mdContactName',
+        contactImage: '@mdContactImage',
+        contactEmail: '@mdContactEmail',
+        filterSelected: '=',
```
Comment (ThomasBurleson) [inline src/components/chips/js/contactChipsDirective.js]: should this be `md-filter-selected`

**Comment by ThomasBurleson** [comment]
I will merge for now. In a separate PR, can you fix `md-filter-selected` so it is consistently used and documented?

**Pull request closed without merging**